pkg/util/helper/unstructured_test.go

```diff
@@ -0,0 +1,194 @@
+package helper
+
+import (
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func TestConvertToTypedObject(t *testing.T) {
+	testCases := []struct {
+		name        string
+		in          interface{}
+		out         interface{}
+		expectedErr bool
+	}{
+		{
+			name:        "in object is nil",
+			in:          nil,
+			expectedErr: true,
+		},
+		{
+			name:        "out object is nil",
+			in:          &unstructured.Unstructured{},
+			out:         nil,
+			expectedErr: true,
+		},
+		{
+			name:        "in object with invalid type",
+			in:          "some",
+			out:         &corev1.Pod{},
+			expectedErr: true,
+		},
+		{
+			name: "convert unstructured object success",
+			in: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"metadata": map[string]interface{}{
+						"name": "some-pod",
+					},
+				},
+			},
+			out:         &corev1.Pod{},
+			expectedErr: false,
+		},
+		{
+			name: "convert unstructured object success",
```

Comment:
looks like it's duplicated with the previous one.

Reply:
Thanks for your review, remove the duplicated case